src/compiler/checker.ts

```diff
@@ -14507,7 +14507,7 @@ namespace ts {
                                 if (patternElement.kind !== SyntaxKind.OmittedExpression) {
                                     error(patternElement, Diagnostics.Initializer_provides_no_value_for_this_binding_element_and_the_binding_element_has_no_default_value);
                                 }
-                                elementTypes.push(unknownType);
+                                elementTypes.push(emptyObjectType);

```

Comment:
What about `strictNullChecks ? silentNeverType : undefinedWideningType`? Similar to what we do several lines below.

Reply:
Mmm, good call. I think I like that behavior more.